repose-aggregator/core/repose-core/src/test/scala/org/openrepose/powerfilter/ReposeFilterChainTest.scala

```diff
@@ -295,17 +316,19 @@ class ReposeFilterChainTest extends FunSpec with Matchers with MockitoSugar with
     }
 
     it("should create a span with the filter name") {
-      val filterChain = new ReposeFilterChain(List(FilterContext(mockFilter, "foo", (request: HttpServletRequest) => true),
-                                                   FilterContext(mock[Filter], "bar", (request: HttpServletRequest) => true)),
-                                              originalChain,
-                                              None,
-                                              metricsRegistry,
-                                              tracer)
+      val filterChain = new ReposeFilterChain(
+        List(
+          FilterContext(mockFilter, "foo", (request: HttpServletRequest) => true),
+          FilterContext(mock[Filter], "bar", (request: HttpServletRequest) => true)),
+        originalChain,
+        None,
+        metricsRegistry,
+        tracer)
 
       filterChain.doFilter(mockRequest, mockResponse)
 
       tracer.finishedSpans should have length 1
-      tracer.finishedSpans.asScala.head.operationName shouldBe "Filter foo"
+      tracer.finishedSpans.asScala.headOption.map(_.operationName) shouldBe Some("Filter foo")
```

Comment:
I wonder if failure message here could be improved using a couple more pieces of ScalaTest, like so:

```
import org.scalatest.OptionValues._
...
tracer.finishedSpans.asScala.loneElement should have (operationName ("Filter foo"))
```

The `loneElement` syntax would let you drop the length check, too.

I would actually be very tempted to use `JavaConversions` here to clean that line up even more (especially given that this is test code), but alas, it is deprecated.

Oh, and as a side note, when working with `Option`s, I would recommend using the `Option(...).value shouldBe X` syntax where possible.

References:
http://www.scalatest.org/user_guide/using_matchers#singleElementCollections
http://www.scalatest.org/user_guide/using_matchers#checkingArbitraryProperties
http://doc.scalatest.org/3.0.1/#org.scalatest.matchers.HavePropertyMatcher

http://www.scalatest.org/user_guide/using_matchers#workingWithOptions

Reply:
I updated this one and added a story to fix the rest:
* https://repose.atlassian.net/browse/REP-7179